app/uk/gov/hmrc/helptosave/models/ErrorType.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.models
+
+import play.api.libs.json._
+
+sealed trait ErrorType
+
+case object Eligibility extends ErrorType
+
+case object UserInfoType extends ErrorType
+
+case object UnKnown extends ErrorType
+
+object ErrorType {
+
+  implicit val ErrorTypeFormat: Format[ErrorType] = new Format[ErrorType] {
+    def reads(json: JsValue): JsResult[ErrorType] = {
+      val errorType = json match {
+        case JsString("Eligibility") ⇒ Eligibility
+        case JsString("UserInfoType") ⇒ UserInfoType
+        case JsString(_) ⇒ UnKnown
```

Comment:
Do you really want to catch everything else as an `Unknown`? It's safer to have a case for `JsString("Unknown")` and have the other case as a `JsError`